targettests/execution/cast.cpp

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// clang-format off
+// RUN: nvq++ %cpp_std --enable-mlir %s -o %t && %t | FileCheck %s
+// RUN: nvq++ %cpp_std --target infleqtion      --emulate %s -o %t && %t | FileCheck %s
```

Comment:
Since this change is generic and applies to any input at all (canonicalization is always run), I think we can conserve some CPU resources here and just execute the test once. We have other ways of tests all the various targets, and this is never going to change wrt the target.
We can get rid of run lines from line 11 through 18.

Reply:
done!